frontend/src/modules/entitydetails/components/Metadata.js

```diff
@@ -0,0 +1,140 @@
+/* @flow */
+
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import Linkify from 'react-linkify';
+
+import './Metadata.css';
+
+import Screenshots from './Screenshots';
+import type { DbEntity } from 'modules/entitieslist';
+
+
+type PropertyProps = {|
+    title: string,
+    children: React.Node,
+|};
+
+/**
+ * Component to dislay a property of an entity.
+ *
+ * Only used by Metadata.
+ */
+class Property extends React.Component<PropertyProps> {
+    render(): React.Node {
+        const { children, title } = this.props;
+        const className = title.trim().toLowerCase().replace(/ /g, '-');
+        return <p className={ className }>
+            <span className="title">{ title }</span>
+            <span className="content">{ children }</span>
+        </p>;
+    }
+}
+
+
+type Props = {|
+    entity: DbEntity,
+    locale: string,
+|};
+
+
+/**
+ * Component showing metadata of an entity.
+ *
+ * Shows:
+ *  - the original string
+ *  - a comment (if any)
+ *  - a context (if any)
+ *  - a list of source files (if any)
+ *  - a link to the resource
+ *  - a link to the project
+ */
+export default class Metadata extends React.Component<Props> {
+    renderComment(entity: DbEntity): React.Node {
+        if (!entity.comment) {
+            return null;
+        }
+
+        let comment = entity.comment;
+
+        const parts = entity.comment.split('\n');
+        if (parts[0].startsWith('MAX_LENGTH')) {
+            // This comment contains a max length instruction. Remove that part.
+            parts.shift();
+            comment = parts.join('\n');
+        }
+
+        return <Property title='Comment'>
+            <Linkify properties={ { target: '_blank', rel: 'noopener noreferrer' } }>
+                { comment }
+            </Linkify>
+        </Property>;
+    }
+
+    renderContext(entity: DbEntity): React.Node {
+        if (!entity.key) {
+            return null;
+        }
+
+        return <Property title='Context'>{ entity.key }</Property>;
+    }
+
+    renderSourceArray(source: Array<Array<string>>): React.Node {
+        return <ul>{ source.map((value, key) => {
+            return <li key={ key }><span className="title">#:</span>{ value.join(':') }</li>;
+        }) }</ul>;
+    }
```

Comment:
Nit: this will return `<ul></ul>` if we don't have any source files instead of `null`.

Reply:
In the case of an empty `source`, that's handled in `renderSources`. But it might be possible that `source = [[]]` in which case we would indeed return an empty list. I've added a check for that.